include/zephyr/net/prometheus/collector.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_PROMETHEUS_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PROMETHEUS_COLLECTOR_H_
+
+/**
+ * @file
+ *
+ * @brief Prometheus collector APIs.
+ *
+ * @defgroup prometheus Prometheus API
+ * @since 3.8
+ * @version 0.1.0
+ * @ingroup networking
+ * @{
+ */
+
+#include <zephyr/sys/iterable_sections.h>
+#include <zephyr/net/prometheus/metric.h>
+
+#include <stddef.h>
+
+/**
+ * @brief Prometheus collector definition
+ *
+ * This structure defines a Prometheus collector.
+ */
+struct prometheus_collector {
+	/** Name of the collector */
+	const char *name;
+	/** Array of metrics associated with the collector */
+	struct prometheus_metric *metric[CONFIG_PROMETHEUS_MAX_METRICS];
+	/** Number of metrics associated with the collector */
+	size_t size;
+};
+
+/**
+ * @brief Prometheus Collector definition.
+ *
+ * This macro defines a Collector.
+ *
+ * @param _name The collector's name.
+ *
+ * @example
+ *
+ * PROMETHEUS_COLLECTOR_DEFINE(test_collector);
```

Comment:
This is not the right use for this command 
https://www.doxygen.nl/manual/commands.html#cmdexample

In any case, this is not really needed as it's not adding any information IMO


```suggestion
```